modules/sfp_dnsdumpster.py

```diff
@@ -0,0 +1,142 @@
+# -*- coding: utf-8 -*-
+# -------------------------------------------------------------------------------
+# Name:        sfp_dnsdumpster
+# Purpose:     SpiderFoot plug-in for subdomain enumeration using
+#              dnsdumpster.com
+#
+# Author:      TheTechromancer
+#
+# Created:     05/21/2021
+# Copyright:   (c) Steve Micallef 2021
+# Licence:     GPL
+# -------------------------------------------------------------------------------
+
+import re
+from bs4 import BeautifulSoup
+
+from spiderfoot import SpiderFootEvent, SpiderFootPlugin
+
+
+class sfp_dnsdumpster(SpiderFootPlugin):
+
+    meta = {
+        "name": "DNSDumpster",
+        "summary": "Passive subdomain enumeration using HackerTarget's DNSDumpster",
+        "useCases": ["Investigate", "Footprint", "Passive"],
+        "categories": ["Passive DNS"],
+        "dataSource": {
+            "website": "https://dnsdumpster.com/",
+            "model": "FREE_NOAUTH_UNLIMITED",
+            "description": "DNSdumpster.com is a FREE domain research tool that can discover hosts related to a domain.",
+        }
+    }
+
+    # Default options
+    opts = {}
+
+    # Option descriptions
+    optdescs = {}
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.sf.debug("Setting up sfp_dnsdumpster")
+        self.state = self.tempStorage()
+        self.state.update({
```

Comment:
I like the approach taken here but for now can you make this consistent with all other modules (using `self.results`)? We'll move to a better state management approach later on.